lxd/container_lxc.go

```diff
@@ -1966,7 +1969,15 @@ func (c *containerLXC) startCommon() (string, error) {
 			if m["path"] != "/" {
 				diskDevices[k] = m
 			}
-		} else if m["type"] == "nic" {
+		} else if m["type"] == "nic" || m["type"] == "infiniband" {
+			var err error
+			if m["type"] == "infiniband" && infiniband == nil {
```

Comment:
Not sure about the caching here unless we also have a uevent hook to clear the cache when we see new devices being added/removed as right now, if I'm doing my usual testing, I'm going to get pretty confused as I usually start with a system that has no visible card, install and start LXD and then setup the functions and stuff.

Reply:
Was only added for performance reasons but sure.

Reply:
Yeah, but that's only helpful if it's right, which it can't be unless you also have cache invalidation :)

Reply:
addressed